engine/crates/parser-sdl/src/rules/mongodb_directive.rs

```diff
@@ -241,4 +243,73 @@ mod tests {
             "Connector names must be alphanumeric and cannot start with a number"
         );
     }
+
+    #[test]
+    fn basic_introspection() {
+        let config = r#"
+extend schema
+  @mongodb(
+    namespace: false
+    name: "MongoDB"
+    url: "https://example.org"
+    apiKey: "gloubiboulga"
+    dataSource: "data-source"
+    database: "database"
+  )
+
+extend schema @federation(version: "2.3")
+
+enum Enum1 {
+  Variant1,
+  Variant2,
+  Variant3
+}
+
+type Model1 @model(connector: "MongoDB", collection: "Model1") @key(fields: "field3") {
+  field1: String!
+  field2: String!
+  field3: String! @unique
+}
+
+type Model2 @model(connector: "MongoDB", collection: "Model2") @key(fields: field1) {
+  field1: String!
+  field2: Enum1!
+  field3: [Model1!]!
+}
+        "#;
+
+        let registry = crate::parse_registry(config).unwrap();
+
+        for (type_name, tpe) in registry.types.iter() {
```

Comment:
This is very primitive. Do we have a better way to check that a registry is consistent re: type references?